### PR TITLE
AppDelegate: Handle unrecognized certificates by prompting user durin…

### PR DESCRIPTION
**Pull request by giomfo**
…g authentication challenge.

### CODE REVIEW

File: matrixConsole/AppDelegate.m [hunk 1]
```diff
@@ -423,6 +425,52 @@ - (void)application:(UIApplication*)application didReceiveRemoteNotification:(NS
...
+        __block BOOL isTrusted;
+        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+        
+        MXCredentials *mxCredentials = mxAccount.mxCredentials;
+        
+        NSString *title = [NSBundle mxk_localizedStringForKey:@"ssl_could_not_verify"];
+        NSString *existing_expl = mxCredentials.allowedCertificate ? [NSBundle mxk_localizedStringForKey:@"ssl_expected_existing_expl"] : [NSBundle mxk_localizedStringForKey:@"ssl_unexpected_existing_expl"];
+        NSString *homeserverURLStr = [NSString stringWithFormat:[NSBundle mxk_localizedStringForKey:@"ssl_homeserver_url"], mxCredentials.homeServer];
+        NSString *fingerprint = [NSString stringWithFormat:[NSBundle mxk_localizedStringForKey:@"ssl_fingerprint_hash"], @"SHA256"];
+        NSString *certFingerprint = [certificate SHA256AsHexString];
```
Comment (dbkr) [inline matrixConsole/AppDelegate.m:439]: This looks like the same code that's in MatrixKit. In what situations are each used and do we need both?

Comment (dbkr) [inline matrixConsole/AppDelegate.m:439]: Oh, I see - this is when the cert changes. Why is one handled by the app and one in MatrixKit?

**Comment by giomfo** [comment]
One block is used to handle unrecognized certificate during first server request which takes place during login/registration step. This block is located at MatrixKit level, because we have the right view controller (MXKAuthenticationViewController). I plan to add a property in this view controller to let the developer customizes the displayed prompt.

The second block handles the certificate change, which may happen at any time during the account use. Only the app delegate may handle correctly this case. That is why this block is defined at App level.

To sum up, the block defined at MatrixKit is here to help developer, but the developer is supposed to define his own block in his MXKAuthenticationViewController-inherited class (For that I need to add the related property (presently missing)).

**Comment by giomfo** [comment]
Change has been done in MatrixKit to let developer override default handling of unrecognized certificate (see MXKAuthenticationViewController class).

**Comment by dbkr** [comment]
LGTM
